### PR TITLE
OMI_Personality

### PR DESCRIPTION
**Pull request by ultrabot2022**
## What Changed:
SXP component has been changed with OMI_Personality extension.

## How to test:
Drag/Drop VRM files to see the agent

### CODE REVIEW

**Comment by fire** [comment]
There are some errors in ESLint. Are you able to resolve?

**Comment by Knar33** [comment]
This is injecting the same personality into every VRM that gets uploaded? Seems like we'd want to grab the data out of them dynamically.

**Comment by ultrabot2022** [comment]
> This is injecting the same personality into every VRM that gets uploaded? Seems like we'd want to grab the data out of them dynamically.

Yes, it is injecting OMI_Personality but they are not same. The Agent is related with node id.

**Comment by ultrabot2022** [comment]
> There are some errors in ESLint. Are you able to resolve?

I just come back here. I will be busy but I will fix it soon.

File: plugins/avatar/client/src/components/Chat/ChatBox.tsx [hunk 1]
```diff
@@ -147,14 +147,15 @@ export default function ChatBox() {
...
     // Stop speech to text when a message is sent through the input
-    stopSpeech()
-    if (!waitingForResponse) {
-      setWaitingForResponse(true)
-      // Get the value of the input element
-      const input = event.target.elements.message
-      const value = input.value
-      handleUserChatInput(value)
-    }
+    // stopSpeech()
```
Comment (fire) [inline plugins/avatar/client/src/components/Chat/ChatBox.tsx:150]: Remove unused code that is commented.

File: plugins/avatar/client/src/components/Chat/ChatBox.tsx [hunk 2]
```diff
@@ -147,14 +147,15 @@ export default function ChatBox() {
...
+
+    // Get the value of the input element
+    const input = event.target.elements.message
+    const value = input.value
+    handleUserChatInput(value)
+    
+    //mic button on/off
```
Comment (fire) [inline plugins/avatar/client/src/components/Chat/ChatBox.tsx:157]: Remove unused code that is commented.

File: plugins/avatar/client/src/components/Scene/XrTest.css [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  
+  html,
+  body,
+  #root {
+    width: 100%;
+    height: 100%;
+    margin: 0;
+    padding: 0;
+  }
+
```
Comment (fire) [inline plugins/avatar/client/src/components/Scene/XrTest.css:13]: Add a new line.

File: plugins/avatar/client/src/hooks/useOMIPersonality.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        "required": ["agent", "personality"]
+      }
+
+      setNewVRM(oldVRM);
+    }
+
+  }, [oldVRM])
+
+  return newVRM
+}
```
Comment (fire) [inline plugins/avatar/client/src/hooks/useOMIPersonality.ts:39]: Add a new line.

**Review by fire — CHANGES_REQUESTED** [review]

File: plugins/avatar/client/src/components/Sxp/Sxp.jsx [hunk 5]
```diff
@@ -1,33 +1,62 @@
...
+    useEffect(()=>{
+      if(newVRM) {
+        try {
+          const gltf = newVRM;
+          // const personalities = [];
+      
+          traverseNodes(gltf.scene, (node) => {
+            // if (node.userData && node.userData.gltfExtensions && node.userData.gltfExtensions.OMI_personality) {
+            if (gltf.scene.userData) {
+              // const omiPersonality = node.userData.gltfExtensions.OMI_personality;
```
Comment (fire) [inline plugins/avatar/client/src/components/Sxp/Sxp.jsx:35]: Why is this code commented?

1. If used please implement
2. Else delete.

File: plugins/avatar/client/src/hooks/useVrmMixamoAnimations.ts [hunk 6]
```diff
@@ -27,9 +26,9 @@ export const useVrmMixamoAnimations = (vrm, mixamoAnimations, exceptionBoneNames
       if (exceptionBoneNames.indexOf(vrmBoneName) > -1) {
         return
       }
-      customDebug().log('useVrmMixamoAnimations#useEffect: vrmBoneName: ', vrmBoneName)
+      // customDebug().log('useVrmMixamoAnimations#useEffect: vrmBoneName: ', vrmBoneName)
       const vrmBoneNode = vrm.humanoid?.getBoneNode(vrmBoneName)
-      customDebug().log('useVrmMixamoAnimations#useEffect: vrmBoneNode: ', vrmBoneNode)
+      // customDebug().log('useVrmMixamoAnimations#useEffect: vrmBoneNode: ', vrmBoneNode)
```
Comment (fire) [inline plugins/avatar/client/src/hooks/useVrmMixamoAnimations.ts:31]: If used please complete, else delete the comment.

File: plugins/avatar/client/src/components/Scene/Scene.jsx [hunk 7]
```diff
@@ -1,43 +1,78 @@
...
+        <ambientLight intensity={0.5} />
 
-      <Suspense>
-        <Physics colliders="hull">
-          <Avatar />
-          <Watch />
-          {/* <Debug/> */}
-        </Physics>
-      </Suspense>
+        {/* <axesHelper args={[AXIS_LEN]} /> */}
```
Comment (fire) [inline plugins/avatar/client/src/components/Scene/Scene.jsx:63]: Why is this code commented?

File: plugins/avatar/client/src/App.module.css [hunk 8]
```diff
@@ -118,3 +117,15 @@
...
+  position: absolute;
+  width: 100%;
+  height: 100%;
+  /* z-index: 1000; */
+  background-color: transparent;
+}
+
+.dropzone-content {
+  color: #fff;
+}
```
Comment (fire) [inline plugins/avatar/client/src/App.module.css:131]: Missing new line at the end of the file.

**Comment by fire** [comment]
VR testing is not requested for now.

**Comment by fire** [comment]
What is the state of LookingGlass support? Do you want it to be tested?

**Pull request closed without merging**